openretina/utils/plotting.py

```diff
@@ -388,3 +400,370 @@ def legend_without_duplicate_labels(ax=None) -> None:
+        )
+
+    # Resize the videos if necessary
+    video1_resized = resize_video(video1, max_height) if bigger_index == 2 else video1
+    video2_resized = resize_video(video2, max_height) if bigger_index == 1 else video2
+
+    # Create a white band of the desired width
+    band = 255 * np.ones((time, max_height, separator_width, channels), dtype=np.uint8)
+
+    # Concatenate the videos with the band in between
+    stitched_video = np.concatenate((video1_resized, band, video2_resized), axis=2)
+    return stitched_video
+
+
+def create_roi_animation(
+    roi_mask: Int[np.ndarray, "roi_height roi_width"],
+    activity: Float[np.ndarray, "n_neurons time"],
+    roi_ids: Int[np.ndarray, " n_neurons"] | None = None,
+    cell_types: Int[np.ndarray, " n_neurons"] | None = None,
+    min_cell_type: int | None = 1,
+    max_cell_type: int | None = 46,
+    type_boundaries: list[int] | None = None,
+    max_activity: int = 10,
+    visualize_ids: bool = False,
+    figsize: tuple[int, int] = (8, 8),
+) -> Float[np.ndarray, "time fig_height fig_width 3"]:
+    """
+    Create an animation of neuronal activity and return a numpy tensor of video frames.
+
+    ---
+    Parameters:
+    roi_mask : np.ndarray
+        2D array where background is 1 and neurons are -n (n being neuron index)
+    activity : np.ndarray
+        n x time array of neuronal activity
+    roi_ids : np.ndarray
+        1D array indicating the neuron ID for each row in activity (default None)
+    cell_types : np.ndarray
+        1D array indicating the cell type for each neuron (default None)
+    type_boundaries : list[int]
+        List of integers indicating boundaries between broad cell types
+    max_activity : float
+        Maximum activity value for scaling (default 10)
+    visualize_ids : bool
+        If True, displays neuron IDs on their ROIs
+    figsize : tuple
+        Figure size in inches (width, height), default (8, 8)
+
+    ---
+    Returns:
+    np.ndarray
+        Numpy array with shape (time, height, width, channels) containing video frames.
+    """
+
+    # Get number of neurons and normalize activity
+    n_neurons = np.min(roi_mask) * -1
+    activity_normalized = np.clip(activity / max_activity, 0, 1)
+
+    # Pre-compute color mapping if cell types are provided
+    color_map = {}
+    if cell_types is not None:
+        max_type = np.max(cell_types) if max_cell_type is None else max_cell_type
+        min_type = np.min(cell_types) if min_cell_type is None else min_cell_type
+
+        # Create broad type ranges
+        if type_boundaries is None:
+            type_boundaries = []
+        type_boundaries = sorted([min_type] + type_boundaries + [max_type])
+
+        # For each broad type range, assign a base saturation
+        saturations = {}
+        for i in range(len(type_boundaries) - 1):
+            start, end = type_boundaries[i], type_boundaries[i + 1]
```

Comment:
```suggestion
        for start, end in zip(type_boundaries, type_boundaries[1:]):
```
this is easier and shorter I think.